### DISCUSSION TITLE
Delegate, Action, Func, Lambda expression - What the heck!?

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#67] [General]
# Delegate, Action, Func, Lambda expression - What the heck!?

C# offers a lot of utility especially around the delegate topic. So let's see what exactly a delegate is and how the distinct types like delegate, Action, Func, Predicate, anonymous function,  lambda expressions and MulticastDelegate behave. A lot to digest and discover so let's go.

https://steven-giesel.com/blogPost/aa77f7fb-41a7-47d2-8e48-ef9e101cf08e

**Comment by PrashantUnity** [comment]
Thankyou So Much

**Comment by fisnik97** [comment]
Amazing explanation! Thanks a lot for putting forward detailed explanations along with usage ideas of each delegate type.
